src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ProjectState.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Testing
+{
+    public abstract class ProjectState
+    {
+        private readonly string _name;
+        private readonly string _defaultPrefix;
+        private readonly string _defaultExtension;
+
+        protected ProjectState(string name, string defaultPrefix, string defaultExtension)
```

Comment:
nit: Perhaps call this `ProjectData``?

Reply:
nevermind this is to match `SolutionState`.

Reply:
Yes. I have a good feeling this API will change forms (e.g. to support diagnostics/markup/Fix All in Solution testing), but I'm hoping it's a decent start to unblock some testing for @mavasani.